templates/utils.ts

```diff
@@ -30,3 +31,33 @@ export function filesToWebContainerFs(
   }
   return tree
 }
+
+export function filesToVirtualFsTree(
+  files: VirtualFile[],
+) {
+  const tree: VirtualFileSystemTree = {}
+
+  for (const file of files) {
+    if (!file.filepath.includes('/')) {
+      tree[file.filepath] = { file }
+    }
+    else {
+      const parts = file.filepath.split('/')
+      const filename = parts.pop()!
+      let current = tree
+      for (const dir of parts) {
+        if (!current[dir]) {
+          current[dir] = {
+            directory: {},
+          }
+        }
+        const node = current[dir]
+        if (!('directory' in node))
+          throw new Error('Unexpected directory but found file')
+        current = node.directory
+      }
+      current[filename] = { file }
+    }
+  }
+  return tree
+}
```

Comment:
I think the code here (`filesToVirtualFsTree` function) is somewhat repetitive above, but I'm not sure whether it should be merged with the functionality above, or if the whole approach should be redesigned.

Reply:
I am ok with keeping them as is, as the logic is basically creating trees and we do need to over-refactor our code before we need that.